src/rez/pip.py

```diff
@@ -267,16 +268,30 @@ def pip_install_package(source_name, pip_version=None, python_version=None,
         tools = []
         src_dst_lut = {}
 
-        for installed_file in distribution.list_installed_files(allow_fail=True):
-            source_file = os.path.normpath(os.path.join(destpath, installed_file[0]))
+        for installed_file in distribution.list_installed_files():
+            # a relative directory that needs to be appended without the
+            # os.sep (via os.path.join) in order for normpath to return the
+            # correct path, this is in order to match the wheel RECORD file
+
+            # Example
+
+            # destpath = /tmp/pip-HBkRGa-rez/re]z_staging/python
+            # installed_file[0] = ../../bin/pydocstyle
+            # normpath of /tmp/pip-HBkRGa-rez/rez_staging/python../../bin/pydocstyle =  /tmp/pip-HBkRGa-rez/rez_staging/bin/pydocstyle OK
+            # normapth of /tmp/pip-HBkRGa-rez/rez_staging/python/../../bin/pydocstyle = /tmp/pip-HBkRGa-rez/bin/pydocstyle WRONG
+            if installed_file[0].startswith("."):
+                installed = destpath + installed_file[0]
+            else:
+                installed = os.path.join(destpath, installed_file[0])
+
+            source_file = os.path.normpath(installed)
 
             if os.path.exists(source_file):
-                destination_file = installed_file[0].split(stagingsep)[1]
+                destination_file = source_file.split(stagingsep)[1]
```

Comment:
I apologise for my own (old) rubbish code! This split is odd, simple use of os.path.relpath would be clearer

```
destination_file = os.path.relpath(source_file, stagingdir)
```